tsdb/engine/tsm1/compact.go

```diff
@@ -176,6 +176,13 @@ func (c *Compactor) writeNewFiles(iter KeyIterator) ([]string, error) {
 		if err == errMaxFileExceeded {
 			files = append(files, fileName)
 			continue
+			// If the file only contained tombstoned entries, then it would be a 0 length
```

Comment:
`go fmt` formatting oddity.  It's for the lines just below it.  E.g. if we iterated through all the files and we actually didn't write anything (because all the keys were tombstoned), we delete the temp file here so we don't end up with a 0 length tsm file to install.